src/cli/self_update/windows.rs

```diff
@@ -616,11 +603,9 @@ pub(crate) fn do_add_to_programs(process: &Process) -> Result<()> {
     let key = rustup_uninstall_reg_key()?;
 
     // Don't overwrite registry if Rustup is already installed
-    let prev = key
-        .get_raw_value("UninstallString")
-        .map(|val| from_winreg_value(&val));
-    if let Ok(Some(s)) = prev {
-        let mut path = PathBuf::from(OsString::from_wide(&s));
+    let prev = key.get_hstring("UninstallString");
+    if let Ok(s) = prev {
+        let mut path = PathBuf::from(s.to_os_string());
```

Comment:
I see we're also converting from `HSTRING` to `PathBuf`.

Reply:
`PathBuf` doesn't implement `From<HSTRING>`, so I have to use this conversion temporarily.

Reply:
You can likely coerce the PathBuf to Path.